docs/src/code/style-guide.adoc

```diff
@@ -252,16 +282,126 @@ of .c and .h are reserved for plain C. Headers are for class, method,
 and structure declarations, not code (unless the functions are declared
 inline).
 
-== Python coding standards
+=== Python coding standards
 
 Use the http://www.python.org/dev/peps/pep-0008/[PEP 8] style for
 Python code.
 
-== Comp coding standards
+=== Comp coding standards
 
 In the declaration portion of a .comp file, begin each declaration at
 the first column. Insert extra blank lines when they help group related
 items.
 
 In the code portion of a .comp file, follow normal C coding style.
 
+== Code describing the functionality for the human
+
+This is a very recent (1/2022) part of this document. Please help shaping it if you are familiar with asciidoc.
+
+After the website, and maybe the one or other YouTube Video, the
+LinuxCNC documentation is likely the first point of contact for any new
+user. The talent to get the documentation right is mostly disjunct from
+computationally orchestrating all the moving parts of a mill or lathe,
+though. Still, we need this to shine if we want LinuxCNC to shine and
+for a transfer of our knowledge for the next generation - this seems
+fair to say for a project that was started in the last millenium and few
+individuals buying their own mills/lathes before their hair turns gray.
+
+=== Overall structure of documentation
+
+There are two basic documents, i.e. the
+ * Users' Guide and the
+ * Developers' Guide
+
+All documents created belong to either of these "parental documents"
+and are included from the respective document, either directly or that
+included file includes it.
+
+=== Common challenges
+
+ * header tags block like toc etc
+   
+   Every file should start with a header. This is typically
+   ----
+   :lang: en
+   ----
+   and longer documents may also chose to set
+   ----
+   :toc:
+   ----
+   This should be followed by an anchor for that section or chapter that is
+   represented in that file, like
+   ----
+   [[cha:filename]]
+   ----
+   If the file has a language-extension like _fr then this extension shall not become part of the anchor.
+   For a better consistency, and maintainability, the anchors need to be the same for all langages.
+   If the page needs extras, e.g. for the syntax-highlighting, then such lines may then follow.
+
+ * anchors for titles and other blocks
+
+   If a chapter/section header shall be granted the option to be
+   referenced from another part of the documentation then it needs an achor.
+   The anchor shall be a combination of an indicator of the kind
+   of block that is referenced (cha,sec,fig,tab,...) together with a
+   short name identifying the object.
+   
+ * index entries for titles and other blocks
+
+   ?
```

Comment:
I'd be in favor of indexing every chapter/section header, as well as every special/rich content blocks like tables, figures,...
Past these, subs of special interest should also be indexed.
Using sub entries, a nicely structured extensive index could be built.

Reply:
This refers to https://docs.asciidoctor.org/asciidoc/latest/sections/user-index/ and the (((names))) specified throughout the texts.
How do we deal with translations? Leave the English and have translations as secondary indices? Vice versa? Only the Translation?

Reply:
Yes it does.
For the most part, I think we should have only translations. For specific and/or hardly translatable entries, a proper use of "see also".
Found https://web.archive.org/web/20180115025406/http://chimera.labs.oreilly.com:80/books/1234000001578/ch02.html#expand a good reference.

Reply:
Thinking about it, we may not have an opportunity to decide about translations - weblate either picks it up or it doesn't.

Reply:
You're right indeed... On second thought, it's more probably a po4a thing, Weblate only seeing what the first has decided to put in po files.